File: 122.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution{
public:
	int maxProfit(vector<int>& prices){
		int ans=0;
		for(int i=0;i+1<prices.size();++i){
			if(prices[i+1]>prices[i])
				ans+=prices[i+1]-prices[i];
		}
		return ans;
	}
};

int main(){
	vector<int> test={7,1,5,3,6,4};
	Solution s;

	cout<<s.maxProfit(test)<<endl;

	return 0;
}

File: Python/122.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        i = 0
        profit = 0
        while i< len(prices):
            j = i+1
            while j<len(prices):
                if prices[j] >= prices[j-1]:
                    j+=1
                else:
                    break
            if prices[j-1]<=prices[i]:
                i+=1
            else:
                profit += prices[j-1]-prices[i]
            i = j
        return profit

class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        profit = 0
        for i in range(1, len(prices)):
            if prices[i] > prices[i-1]:
                profit += prices[i]-prices[i-1]
        return profit
